op-chain-ops/cmd/check-l2/main.go

```diff
@@ -0,0 +1,624 @@
+package main
+
+import (
+	// "errors"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/mattn/go-isatty"
+	"github.com/urfave/cli/v2"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/util"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var defaultCrossDomainMessageSender = common.HexToAddress("0x000000000000000000000000000000000000dead")
+
+// Default script for checking that L2 has been configured correctly. This should be extended in the future
+// to pull in L1 deploy artifacts and assert that the L2 state is consistent with the L1 state.
+func main() {
+	log.Root().SetHandler(log.StreamHandler(os.Stderr, log.TerminalFormat(isatty.IsTerminal(os.Stderr.Fd()))))
+
+	flags := []cli.Flag{}
+	flags = append(flags, util.ClientsFlags...)
+	flags = append(flags, util.AddressesFlags...)
+
+	app := &cli.App{
+		Name:  "check-l2",
+		Usage: "Check that an OP Stack L2 has been configured correctly",
+		Flags: flags,
+		Action: func(ctx *cli.Context) error {
+			clients, err := util.NewClients(ctx)
+			if err != nil {
+				return err
+			}
+
+			log.Info("Checking predeploy proxy config")
+			g := new(errgroup.Group)
+
+			// All admin slots should be set
+			count := uint64(2048)
+			for i := uint64(0); i < count; i++ {
+				i := i
+				if i%4 == 0 {
+					log.Info("Checking proxy", "index", i, "total", count)
+					if err := g.Wait(); err != nil {
```

Comment:
why wait every 4 invocations?

Reply:
It hits rate limits if increased

Reply:
what rate limit? One in our code or on an external RPC?

Reply:
Running this against alchemy